src/lib/EDSL.mli

```diff
@@ -106,26 +116,29 @@ val list_append: 'a list t -> 'a list t -> 'a list t
 
 val list_iter: 'a list t -> f:((unit -> 'a t) -> unit t) -> unit t
 
-val list_to_string: 'a list t -> f:('a t -> string t) -> string t
+val list_to_string: 'a list t -> f:('a t -> byte_array t) -> byte_array t
 
-val list_of_string: string t -> f:(string t -> 'a t) -> 'a list t
+val list_of_string: byte_array t -> f:(byte_array t -> 'a t) -> 'a list t
 
 (** {3 String Manipulation} *)
 
-val output_as_string : unit t -> string t
+val to_byte_array: c_string t -> byte_array t
+val to_c_string: byte_array t -> c_string t
+
+val output_as_string : unit t -> byte_array t
 
-val feed : string:string t -> unit t -> unit t
+val feed : string:byte_array t -> unit t -> unit t
 (** Feed some content ([~string]) into the ["stdin"] filedescriptor of
     a [unit t] expression. *)
 
-val ( >> ) : string t -> unit t -> unit t
+val ( >> ) : byte_array t -> unit t -> unit t
 (** [str >> cmd] is [feed ~string:str cmd]. *)
 
-val string_concat: string t list -> string t
-(** Concatenate an (OCaml) list of [string t] values. *)
+val string_concat: c_string t list -> c_string t
+(** Concatenate an (OCaml) list of [c_string t] values. *)
 
-val string_concat_list: string list t -> string t
-(** Concatenate a Genspio list of strings [string list t]. *)
+val string_concat_list: c_string list t -> c_string t
```

Comment:
Maybe it is time to wrap some of these functions into a List module like with Integers above?

Reply:
Yes I plan to revamp the API quite a bit but not in this PR.